### ISSUE TITLE
Xcode 12.0.1 - Cannot build project due to a deprecation on GoogleDataTransport

### ISSUE THREAD

**Issue by rodmoura** [firebase/firebase-ios-sdk#6694]
### [REQUIRED] Step 1: Describe your environment

  * Xcode version: 12.0.1
  * Firebase SDK version: 6.34.0
  * Firebase Component: Core (Auth, Core, Database, Firestore, Messaging, Storage, etc)
  * Component version: 6.10.4
  * Installation method: `x CocoaPods | Carthage | Zip file | Swift Package Manager` (select one)

#### Steps to reproduce:

What happened? How can we make the problem occur?
This could be a description, log/console output, etc.

After running `pod update`, I am unable to build the project. Getting this error:

`/.../Pods/GoogleDataTransport/GoogleDataTransport/GDTCORLibrary/GDTCORPlatform.m:140:55: 'currentRadioAccessTechnology' is deprecated: first deprecated in iOS 12.0`

`GoogleDataTransport` version is `7.5.0`

If you have a downloadable sample project that reproduces the bug you're reporting, you will
likely receive a faster response on your issue.

#### Relevant Code:

```
//TODO
```

**Comment by angelo-RSVP** [comment]
Same

**Comment by fuxx** [comment]
Me too

**Comment by garfieldmypet** [comment]
Hit this error: 'currentRadioAccessTechnology' is deprecated: first deprecated in iOS 12.0 . 

Error happened in this function: 
GDTCORNetworkMobileSubtype GDTCORNetworkMobileSubTypeMessage() 

We also cannot compile. Please help, we have been debugging for a day.

**Comment by kcw-grunt** [comment]
Very similar:
## Environment:
- Xcode 12.0.1
- Target iOS 12.0
- Linker Flag *on* `Obj-C`

## Partial Log:
```
GoogleAppMeasurement(APMMeasurement_06d557ff791c9b39ab6f619db2079b01.o)
      -[APMMeasurement networkRemoteConfigFetchCompletionHandler:data:error:] in GoogleAppMeasurement(APMMeasurement_06d557ff791c9b39ab6f619db2079b01.o)
      -[APMETaskManager handleFetchingExperimentsResponse:data:error:] in GoogleAppMeasurement(APMETaskManager_b9f0e011ea705d8ed1303f004b31599b.o)
      ___46-[APMMeasurement(URL) requestDeferredDeepLink]_block_invoke in GoogleAppMeasurement(APMMeasurement+URL_ba0ae0fbae4b5ddbbf1aff0fe935b312.o)
  "_OBJC_CLASS_$_GULNetwork", referenced from:
      objc-class-ref in GoogleAppMeasurement(APMAnalytics_77b3934e62fa06935ddca85ccc80b7e8.o)
      objc-class-ref in GoogleAppMeasurement(APMMeasurement_06d557ff791c9b39ab6f619db2079b01.o)
      objc-class-ref in GoogleAppMeasurement(APMAppDelegateInterceptor_37e5ef4296f5ff8ca5ec5d6ec22a791e.o)
      objc-class-ref in GoogleAppMeasurement(APMETaskManager_b9f0e011ea705d8ed1303f004b31599b.o)
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```
|Error Screenshot|
|------------------|
|<img width="860" alt="Screen Shot 2020-10-08 at 12 04 53 PM" src="https://user-images.githubusercontent.com/2899463/95450702-95d0d880-095e-11eb-9936-c9c34458cd24.png">|

**Comment by kcw-grunt** [comment]
UPDATE: Added `config.build_settings.delete IPHONEOS_DEPLOYMENT_TARGET` to match project targeted iOS version (iOS 12) and now it is exactly the same error  filed by @rodmoura

**Comment by ryanwilson** [comment]
Sorry for the issues folks - we're working on a fix and tracking it in #6682 since it was reported earlier. I'll close this so we can track it all in one place, thanks for your patience. You can find a workaround in the comments: https://github.com/firebase/firebase-ios-sdk/issues/6682#issuecomment-704998269
